src/java/org/apache/cassandra/service/StorageServiceMBean.java

```diff
@@ -790,8 +790,15 @@ public StageConcurrency(int corePoolSize, int maximumPoolSize)
     /** Clears the history of clients that have connected in the past **/
     void clearConnectionHistory();
     public void disableAuditLog();
-    public void enableAuditLog(String loggerName, Map<String, String> parameters, String includedKeyspaces, String excludedKeyspaces, String includedCategories, String excludedCategories, String includedUsers, String excludedUsers) throws ConfigurationException;
-    public void enableAuditLog(String loggerName, String includedKeyspaces, String excludedKeyspaces, String includedCategories, String excludedCategories, String includedUsers, String excludedUsers) throws ConfigurationException;
+    public void enableAuditLog(String loggerName, Map<String, String> parameters, String includedKeyspaces, String excludedKeyspaces, String includedCategories, String excludedCategories,
+                               String includedUsers, String excludedUsers, Integer maxArchiveRetries, Boolean block, String rollCycle,
+                               Long maxLogSize, Integer maxQueueWeight, String archiveCommand) throws ConfigurationException, IllegalStateException;
+
+    public void enableAuditLog(String loggerName, Map<String, String> parameters, String includedKeyspaces, String excludedKeyspaces, String includedCategories, String excludedCategories,
+                               String includedUsers, String excludedUsers) throws ConfigurationException, IllegalStateException;
+
+    public void enableAuditLog(String loggerName, String includedKeyspaces, String excludedKeyspaces, String includedCategories, String excludedCategories,
+                               String includedUsers, String excludedUsers) throws ConfigurationException, IllegalStateException;
```

Comment:
About the MBean... it is a good question.
I am not sure how do we act in those cases. Probably the options are two - we leave it "as is" for now and maybe follow up in additional ticket.
Or deprecate them with some backward compatibility? Similar to what we did with some metrics. Also, I am not sure that this could be done in a minor version. CC @michaelsembwever  as I feel he might have more experience with that than me. Mine is more of a personal thoughts than project experience

Reply:
It was / is used in 4.0.0. But I kept it there because of the "compatibility". There is a rule I am following that there is some "unwritten contract" among devs (e.g. ask Markus Ericsson) that if we add a method we just can not remove the one which was there because we might disrupt the behaviour people rely on. So the common practice is to at least annotate it as obsolete (I can do that) and then it will be removed ... sometimes.

Reply:
I deprecated methods which are not accepting new configuration options we are exposing (these low-level ones for bin log as it is done for fql)

I keep them there until the next release, after that I remove these deprecated methods and I move the ones which left in StorageService to new AuditLogManagerMBean.

Reply:
I think that will be done with alias. For reference, I found this good example today https://issues.apache.org/jira/browse/CASSANDRA-16404?page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel&focusedCommentId=17400403#comment-17400403